### ISSUE TITLE
`prefer-math-trunc`: it's not always safe to remove the `String()` wrapper

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#3719]
The suggestion of `prefer-math-trunc`'s report to replace `Number.parseInt(String(x), 10)` with `Math.trunc` substitutes `x`, the `String()` wrapper argument, instead of the whole expression, assuming `x` is always `number`, but it might be not. Passing non-`number`s to `Math.trunc` results in a type error.

Example:

```ts
type LocationQueryValue = string | null;
type LocationQuery = Record<string, LocationQueryValue | LocationQueryValue[]>;

declare const query: LocationQuery;

Number.parseInt(String(query['page'] || '1'), 10)
```

[Open in ESLint playground](https://eslint-online-playground.netlify.app/#eNqFkktv1DAUhf+KlY1nRnlMFwgpBYQEGyQEAiQ241mY5Ca4OI7xA7WayX/n+qFOp52qm8TK+c691+fmUFjTNXDLJy2hdrZoC3engXyeO+7ErL55MHc/ufRA3hLrjFAjORLlpbxm6imJ0HfoZtO/SWx5qdDxwsfd/h0WZKqHTnIDpJuVdeRv0NtzPGJf/PQLTK25sfBJudWP2G0V+R3VfAS6J8cjoVd0XZKr7booC7BSKFdj5UGM9U24qpj0bBw5kB4GoeBDlMhCBjNPhGVHkxyswM7Z4JXAS6rE0YRVWvpRqCpr9EQ7m8GQl+2M0K5KngAxBbcRwxm4l/F9P8tqh4A7hAdzg5BgW7Kjm02zqQ83tnR2ofsyqak96plmLk+S9SW/JVejx4C+6hDpQz6maVqcNwVrHjmNj/3veZobNNrAAKaauPtdIaU62mIqxsyGnpUIz/36GnehefcHZ8AtzAr3gCUJ5h3yYUWLp2nusRkr0BCEHv59BA34d6hOgA1MtAQtJZlskjuweI62k/hoNYl9/are1tsH7Psn62lSDM/UvsCftXvGdnK9COSy51zAFqaWYvkPASlTvQ==)

The suggestion changes the last line to:

```ts
Math.trunc(query['page'] || '1')
```

Which [doesn't typecheck](https://www.typescriptlang.org/play/?ssl=6&ssc=33&pln=6&pc=1#code/C4TwDgpgBAMg9gYwIbAJZwHYEUCuEBOIAakgDZ5QC8UAzsPqhgOZQA+UGOppA3AFChIsRCnTY8hKlABKEBHHwATADx0GzADTDkaTLgLEyFdvB1j9hEuQgBtALoA+fn0VzSSfNHkY6UAI4SIABc2qJ6gc4AsigAFgB09DgYCAAUAQY2AORgSEwQmXZs7JkAjJkAlEA).

**Comment by andreww2012** [comment]
Sorry, ignore this - it's late and for some reason I completely missed that preserving the String wrapper would result in a type error too 🤦‍♂️. I don't think suggestion should guarantee a safe transformation.
